src/lib/codeutil.ts

```diff
@@ -149,4 +156,30 @@ export async function processSourceCode(dataArray){
     };
   });
   return OrderedIssuesArray;
-}
\ No newline at end of file
+}
+
+export function processSuppression(suppression: any) {
+  if (!suppression) return null;
+
+  return {
+    justification: suppression.justification,
+    category: suppression.properties?.category || 'unknown',
+    expiration: suppression.properties?.expiration,
+    ignoredOn: suppression.properties?.ignoredOn,
+    ignoredBy: suppression.properties?.ignoredBy,
+  };
```

Comment:
**issue:** We should add explicit fallback values here for each of the optional values. This would be particularly useful for the ignoredBy property which is an object we use in templates. For example: 

```
ignoredBy: suppression.properties?.ignoredBy || {
  name: 'unknown',
  email: '?'
}
```

Alternatively we could return null for ignoredBy and then the templates should guard against this scenario by gating each `ignoredBy.propertyName` lookup with a `#if suppression.ignoredBy`